### PR TITLE
Gamified fix logic with is complete issue fix 2.0

### PR DESCRIPTION
**Pull request by loui981c**
## Description

Addressing the problem where only exercise components inside the swiper had isComplete set to true has been resolved. Now, all components are updated, and the swiper starts at the first incomplete component, if any, instead of the start. Additionally, there is a refactoring of the points logic, but the underlying logic remains unchanged.

## Changes

The changes are related to issue from mobile https://github.com/Educado-App/educado-mobile/issues/199

## Related Issues

Issue https://github.com/Educado-App/educado-mobile/issues/199

## Checklist

- [X] Code has been tested locally and passes all relevant tests.
- [X] Documentation has been updated to reflect the changes, if applicable.
- [X] Code follows the established coding style and guidelines of the project.
- [X] All new and existing tests related to the changes have passed.
- [X] Any necessary dependencies or new packages have been properly documented.
- [X] Pull request title and description are clear and descriptive.
- [X] Reviewers have been assigned to the pull request.
- [X] Any potential security implications have been considered and addressed.
- [X] Performance impact of the changes has been evaluated, if relevant.

## Notes for Reviewers

I have commented out some of the tests since the functions do not exist and they can't be found in the current dev branch. Please test with a new user and the third section of the course Finance

### CODE REVIEW

**Comment by benceisonline** [comment]
this should be merged before the corresponding mobile merge, every test passes and everything works correctly 🥳 🥳 🥳

File: db/index.js [hunk 1]
```diff
@@ -6,8 +6,6 @@ function connectToDb(uri, options = {}) {
 
 	const db = mongoose.connection;
 	db.on('error', console.error.bind(console, 'MongoDB connection error'));
-
-	_db = db;
```
Comment (Jatewo) [inline db/index.js:10]: Is this supposed to be deleted?

File: helpers/completing.js
Comment (Jatewo) [inline helpers/completing.js:1]: I don't think there's a single test for these... We should test utility functions

File: package.json [hunk 2]
```diff
@@ -18,7 +18,7 @@
     "docker_down": "docker-compose down",
     "docker_build": "docker-compose up -d --build",
     "docker_clean_db": "docker-compose down && docker volume rm educado-backend_mongodb_data && docker-compose up -d --build mongodb",
-    "linting": "eslint --ext .js,.jsx,.ts,.tsx .",
+    "linting": "eslint --ext .js,.jsx,.ts,.tsx . --fix",
```
Comment (Jatewo) [inline package.json:21]: I'm not sure if we should have --fix forced here. Thoughts?

File: routes/studentRoutes.js
Comment (Jatewo) [inline routes/studentRoutes.js:1]: Are you sure that putting something after `:id` in the URLs won't give problems? I am not sure but I believe someone talked about this.

**Review by Jatewo — CHANGES_REQUESTED** [review]
### Nice work

Some small things to address above 😄

File: db/index.js (hunk 1 above)
Comment (benceisonline) [reply db/index.js:10]: yes :)

File: package.json (hunk 2 above)
Comment (sofiiagran) [reply package.json:21]: yes we should @Jatewo , then it runs fix when you call the linting command

File: helpers/answerExercises.js
Comment (sofiiagran) [inline helpers/answerExercises.js:1]: are you meant to delete this?

**Review by sofiiagran — APPROVED** [review]
I approve, just talk to the web group about the login stuff

File: helpers/answerExercises.js
Comment (loui981c) [reply helpers/answerExercises.js:1]: jeps, its moved to completing.js

File: helpers/completing.js
Comment (loui981c) [reply helpers/completing.js:1]: there is in student route since its used there

File: routes/studentRoutes.js
Comment (loui981c) [reply routes/studentRoutes.js:1]: it is outdated, it is :id/info now

File: helpers/completing.js
Comment (Jatewo) [reply helpers/completing.js:1]: That is not a unit test of the utility function

File: package.json (hunk 2 above)
Comment (Jatewo) [reply package.json:21]: I know what it does. But I'm just not sure if we should force it to run with --fix. Now we can't use it to simply check the state of it.

File: routes/studentRoutes.js
Comment (Jatewo) [reply routes/studentRoutes.js:1]: That doesn’t really answer my question. When you put something after `:id`, could it give problems when matching the URL on a request? I haven't personally experienced it, but I just heard someone else talk about it.

Comment (PrometheusBork) [reply routes/studentRoutes.js:1]: It wont give problems as long as there is not a student route called ":id" only

**Pull request merged**